code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,275 @@
+
+/obj/structure/inflatable/wall
+	name = "inflatable wall"
+	undeploy_path = /obj/item/inflatable/wall
+
+/obj/structure/inflatable/New(location)
+	..()
+	update_nearby_tiles()
+
+/obj/structure/inflatable/Destroy()
+	update_nearby_tiles()
+	..()
+
+/obj/structure/inflatable/Cross(atom/movable/mover, turf/target, height=0, air_group=0)
+	return 0
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/projectile_check()
+	return PROJREACT_WINDOWS
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			deflate(1)
+		if(3)
+			if(prob(50))
+				deflate(1)
+
+/obj/structure/inflatable/attack_hand(mob/user)
+	add_fingerprint(user)
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W, mob/user)
+	if(W.is_sharp())
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(1)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+
+/obj/structure/inflatable/attack_alien(mob/user)
+	user.delayNextAttack(10)
+	if(islarva(user))
+		return
+	visible_message("<span class='danger'>\The [user] rips [src] apart!</span>")
+	deflate(1)
+
+/obj/structure/inflatable/attack_animal(mob/user)
+	user.delayNextAttack(10)
+	var/mob/living/simple_animal/M = user
+	if(M.melee_damage_upper <= 0)
+		return
+	user.visible_message("<span class='danger'>\The [user] attacks [src]!</span>")
+	attack_generic(M, M.melee_damage_upper)
+
+/obj/structure/inflatable/attack_slime(mob/user)
+	user.delayNextAttack(10)
+	if(!isslimeadult(user))
+		return
+	user.visible_message("<span class='danger'>\The [user] glomps [src]!</span>")
+	attack_generic(user, rand(10, 15))
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/attackblob.ogg', 75, 1)
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/CtrlClick()
+	hand_deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("\The [src] rapidly deflates!")
+		var/obj/item/inflatable/torn/R = new /obj/item/inflatable/torn(loc)
+		transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		if(!undeploy_path)
+			return
+		visible_message("\The [src] slowly deflates.")
+		spawn(50)
+			var/obj/item/inflatable/R = new undeploy_path(loc)
+			transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(isobserver(usr) || usr.restrained() || !usr.Adjacent(src))
+		return
+
+	verbs -= /obj/structure/inflatable/verb/hand_deflate
+	deflate()
+
+/obj/structure/inflatable/proc/attack_generic(var/mob/user, var/damage, var/attack_verb)
+	health -= damage
+	if(health <= 0)
+		spawn(1) deflate(1)
+	return 1
+
+/obj/structure/inflatable/door
```

Comment:
No, no, no.

'/obj/machinery/door/inflatable'

All doors should be DOORS, not some weird dooralike that slowly breaks in the code.